conformance/02_push_test.go

```diff
@@ -71,15 +71,23 @@ var test02Push = func() {
 				Expect(err).To(BeNil())
 				location := resp.Header().Get("Location")
 				Expect(location).ToNot(BeEmpty())
-				Expect(resp.StatusCode()).To(Equal(http.StatusCreated))
+				Expect(resp.StatusCode()).To(SatisfyAny(
+					Equal(http.StatusCreated),
+					Equal(http.StatusAccepted),
+				))
+				lastResponse = resp
 			})
 
-			g.Specify("GET request to blob URL from prior request should yield 200", func() {
+			g.Specify("GET request to blob URL from prior request should yield 200 or 404 based on response code", func() {
 				SkipIfDisabled(push)
 				req := client.NewRequest(reggie.GET, "/v2/<name>/blobs/<digest>", reggie.WithDigest(configs[1].Digest))
 				resp, err := client.Do(req)
 				Expect(err).To(BeNil())
-				Expect(resp.StatusCode()).To(Equal(http.StatusOK))
+				if lastResponse.StatusCode() == http.StatusAccepted {
+					Expect(resp.StatusCode()).To(Equal(http.StatusNotFound))
+				} else {
+					Expect(resp.StatusCode()).To(Equal(http.StatusOK))
+				}
```

Comment:
looking good should there be a loop waiting for success after the initial (immediate) not found case?

Reply:
I believe that it will continue to be 404 until the blob is upload via the URL in `Location` header. So basically the upload is considered incomplete if it did not return a `201`

@jonjohnsonjr @dmcgowan - please correct me if I'm mistaken

Reply:
Yeah, we aren't testing the client, we're probing the registry for its behavior... so there won't be any success unless we write some code to take appropriate action based on the registry response.

We _could_ have another test that finalizes the upload with a PUT if lastResponse.StatusCode() == http.StatusNotFound, which would test that the registry is able to receive the payload after a "failed" monolithic upload, but it doesn't seem super necessary... I'd leave that up to Josh.

This looks fine to me.

Reply:
This is tested in the very next test here: https://github.com/opencontainers/distribution-spec/blob/bbb8a94b0bfd7ea93ef257d6326fb4a4962cab51/conformance/02_push_test.go#L94

I think less requests is better, unless there is some issue/danger in having an unused upload session created as part of running these tests.

Reply:
Agreed, that looks good to me.